app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaScreen.kt

```diff
@@ -88,11 +82,39 @@ fun checkIfSharingIsEnabled(featureFlagNotificationViewModel: FeatureFlagNotific
     featureFlagNotificationViewModel.updateSharingStateIfNeeded()
 }
 
+@Composable
+fun ImportMediaContent(
+    unauthorizedViewModel: ImportMediaUnauthorizedViewModel,
+    featureFlagNotificationViewModel: FeatureFlagNotificationViewModel
+) {
+
+    when (featureFlagNotificationViewModel.featureFlagState.fileSharingRestrictedState) {
+        FeatureFlagState.SharingRestrictedState.NO_USER -> ImportMediaLoggedOutContent(
+            featureFlagNotificationViewModel = featureFlagNotificationViewModel,
+            viewModel = unauthorizedViewModel
+        )
+        FeatureFlagState.SharingRestrictedState.RESTRICTED_IN_TEAM -> ImportMediaRestrictedContent(
+            featureFlagNotificationViewModel = featureFlagNotificationViewModel
+        )
+        FeatureFlagState.SharingRestrictedState.NONE -> ImportMediaRegularContent()
+    }
+
+    BackHandler { unauthorizedViewModel.navigateBack() }
+}
+
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun ImportMediaContent(importMediaViewModel: ImportMediaViewModel, featureFlagNotificationViewModel: FeatureFlagNotificationViewModel) {
-    val snackbarHostState: SnackbarHostState = remember { SnackbarHostState() }
-    val searchBarState = rememberSearchbarState()
+fun ImportMediaRestrictedContent(
+    featureFlagNotificationViewModel: FeatureFlagNotificationViewModel,
+    importMediaViewModel: ImportMediaAuthorizedViewModel = hiltViewModel()
+) {
+    val context = LocalContext.current
+    LaunchedEffect(importMediaViewModel.importMediaState.importedAssets) {
+        if (importMediaViewModel.importMediaState.importedAssets.isEmpty()) {
+            context.getActivity()?.let { importMediaViewModel.handleReceivedDataFromSharingIntent(it) }
```

Comment:
Is this a copy/paste leftover or do we need to handle the list of imported for any reason on restricted content case?

Reply:
I decided it could be helpful if/when we implement account switching, but in that case user will see `ImportMediaRegularContent`, so yeah - I'll remove it :)